mapillary_tools/exif_read.py

```diff
@@ -378,20 +415,26 @@ def extract_direction(self) -> T.Optional[float]:
         )
 
     def extract_lon_lat(self) -> T.Optional[T.Tuple[float, float]]:
-        lat = self._extract_alternative_fields(["exif:GPSLatitude"], float)
+        lat_str: T.Optional[str] = self._extract_alternative_fields(
+            ["exif:GPSLatitude"], str
+        )
+        lat: T.Optional[float] = _parse_coord(lat_str)
         if lat is None:
             return None
 
-        lon = self._extract_alternative_fields(["exif:GPSLongitude"], float)
+        lon_str: T.Optional[str] = self._extract_alternative_fields(
+            ["exif:GPSLongitude"], str
+        )
+        lon = _parse_coord(lon_str)
         if lon is None:
             return None
 
         ref = self._extract_alternative_fields(["exif:GPSLongitudeRef"], str)
-        if ref and ref.upper() == "W":
+        if ref and ref.upper() == "W" and lon > 0:
```

Comment:
Correct, the Adobe coords already have the right sign, that's why I added the check on lon/lat > 0 - to avoid multiplying them by -1 again. But yeah, it's hacky, I'll make it more explicit :)

Reply:
Why not separating them? So adoble values don't need to respect the ref tags. See my proposal above.